### PR TITLE
Unify queue slurping - msg and tiles

### PR DESCRIPTION
**Pull request by dennisfrancis**
So that the core message ordering is maintained (by enforcing a single
queue).

Signed-off-by: Dennis Francis <dennis.francis@collabora.com>
Change-Id: I6ba0d3ed13f4c9b08461dd7ae046106aeaaa4c50

* Resolves: # 
* Target version: master 

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by dennisfrancis** [comment]
WIP: Need to test many cases.

File: loleaflet/src/core/Socket.js [hunk 1]
```diff
@@ -273,20 +273,36 @@ L.Socket = L.Class.extend({
 			this._map._docLayer.pauseDrawing();
 		}
 		// console.log2('Slurp events ' + that._slurpQueue.length);
-		for (var i = 0; i < this._slurpQueue.length; ++i) {
+		var queueLen = this._slurpQueue.length;
+		var complete = true;
+		for (var i = 0; i < queueLen; ++i) {
 			var evt = this._slurpQueue[i];
-			try {
-				// it is - are you ?
```
Comment (mmeeks) [inline loleaflet/src/core/Socket.js:279]: ah you removed my joke about being on-message ;-)

File: loleaflet/src/layer/tile/CanvasTileLayer.js [hunk 2]
```diff
@@ -1341,9 +1348,11 @@ L.CanvasTileLayer = L.TileLayer.extend({
 			}, 1000);
 		}
 
-		// paint this tile on canvas.
-		this._painter._tilesSection.paint(tile, undefined, true /* async? */);
-		this._painter.paintOverlayArea(coords);
+		// Don't paint the tile, only dirty the sectionsContainer if it is in the visible area.
+		// _emitSlurpedTileEvents() will repaint canvas (if it is dirty).
+		if (this._painter.coordsIntersectVisible(coords)) {
+			this._painter._sectionContainer.setDirty();
```
Comment (mmeeks) [inline loleaflet/src/layer/tile/CanvasTileLayer.js:1354]: love it =)

**Review by mmeeks — APPROVED** [review]
Looks beautiful =) thanks Dennis!

**Comment by dennisfrancis** [comment]
Added a fix for dialog messages with 'nopng'.

File: loleaflet/src/core/Socket.js (hunk 1 above)
Comment (dennisfrancis) [reply loleaflet/src/core/Socket.js:279]: restored ;)

**Comment by dennisfrancis** [comment]
@tml1024 In this PR I've touched some parts where there is IOS app specific code and I do not have such a device to test unfortunately. Could you review this ?

**Comment by tml1024** [comment]
@dennisfrancis I am not really working on the iOS app at the moment, unclear whether I can spend time on that.

**Comment by tml1024** [comment]
@dennisfrancis OK. Will try to figure out how much time I can spend on that.

**Comment by tml1024** [comment]
(Was told that this has priority now, will get right on it.)

**Comment by tml1024** [comment]
There is one fuzzy hunk in the commits for me:

```
patching file loleaflet/src/control/Control.LokDialog.js
patching file loleaflet/src/control/Control.PartsPreview.js
patching file loleaflet/src/core/Socket.js
patching file loleaflet/src/layer/tile/CanvasTileLayer.js
Hunk #1 succeeded at 138 with fuzz 2.
Hunk #2 succeeded at 1280 (offset 1 line).
Hunk #3 succeeded at 1349 (offset 1 line).
Hunk #4 succeeded at 1394 (offset 1 line).
Hunk #5 succeeded at 1616 (offset 1 line).
Hunk #6 succeeded at 1714 (offset 1 line).
patching file loleaflet/src/layer/tile/TileLayer.js
Hunk #1 succeeded at 665 (offset -9 lines).
Hunk #2 succeeded at 2470 (offset 12 lines).
```

**Comment by tml1024** [comment]
(But I will do the obvious fix for that and then check.)

**Comment by dennisfrancis** [comment]
This PR will need the previous "slurping" related very recent patches in co-6-4.

**Comment by tml1024** [comment]
Building core for iOS apparently now needs to enable the "scripting" bits, I get this when I link the iOS app:


```Undefined symbol: _ScVbaEventsHelper_get_implementation
Undefined symbol: _scriptframe_component_getFactory
Undefined symbol: _ScVbaTextFrame_get_implementation
Undefined symbol: _basprov_component_getFactory
Undefined symbol: _ScVbaGlobals_get_implementation
Undefined symbol: _vbaobj_component_getFactory
```
I need to fixify that first.

Possibly the whole --disable-scripting option can be removed from core, if effectively scripting now is always wanted both on Android and iOS.

**Comment by tml1024** [comment]
Briefly testing, the iOS app still seems to work with these changes.

**Comment by tml1024** [comment]
@dennisfrancis I built against 52add3bae28b1e7cda0234110341f3c6e0b06fdc of cp-6.4, is that new enough?

**Comment by dennisfrancis** [comment]
@tml1024 yes that is the latest.

**Review by tml1024 — APPROVED** [review]

**Comment by mmeeks** [comment]
Lets get them in then =)

**Comment by tml1024** [comment]
Now when I used a correct loleaflet/dist, from current co-6-4, with this PR included, it does seem to work, too. Knock on wood.

**Pull request merged**